### DISCUSSION TITLE
Handlebars expressions in SQL update Query doesn't work

### DISCUSSION THREAD

**Discussion by tobias-lewald** [Budibase/budibase#5059] [Help] [answered]
I have a simple query, that should be executed, when a button is clicked.
Query: 
`INSERT INTO mytable 
    (a, b, c, d)
VALUES
    {{inserts}}
ON DUPLICATE KEY UPDATE
    c = NOW(),
    d = 'disabled'
`
For example my insert binding looks like `(1, 2, NOW(), 'active'), (2, 3, NOW(), 'active')`. When i try to run the query I get an error, but when I copy the binding directly into the query everything works. How can I fix this?

**Comment by mjashanks** [comment, marked as answer]
Hi @tobias-lewald. With SQL bindings, we do not do simple text replacement as you expect, as this would leave the query vulnerable to SQL injection.

You will have to create a binding for ever value in the values clause e.g.

```sql
INSERT INTO mytable (a, b, c, d) 
VALUES ({{a}}, {{b}}, NOW() , '{{d}}')
ON DUPLICATE KEY UPDATE c = NOW(), d = 'disabled'
```

**Reply by tobias-lewald** [reply]
Hi, thanks for the answer. How can I insert a variable number of values? I want to select from two tables entries and insert the cartesian product into another table.

**Reply by mjashanks** [reply]
Hey @tobias-lewald - I don't think this is possible right now. You could maybe pass your values via a stored procedure?
